tests/unit/test_report_junit.py

```diff
@@ -120,11 +120,14 @@ def test_pass(self, report_fix):
 
         report.go()
 
-        assert_xml(out_file_path, """<?xml version="1.0" ?>
+        assert_xml(out_file_path, """<?xml version='1.0' encoding='utf-8'?>
 <testsuites disabled="0" errors="0" failures="0" tests="1" time="0.0">
-    <testsuite disabled="0" errors="0" failures="0" name="name" skipped="0" tests="1" time="0">
-        <testcase name="/pass"/>
-    </testsuite>
+  <testsuite name="name" disabled="0" errors="0" failures="0" skipped="0" tests="1" time="0.0">
+    <testcase name="/pass">
+
```

Comment:
@thrix The data content inside the testcase tags cannot be easily removed - even the blank content still counts as data.

From the lxml [docs](https://lxml.de/tutorial.html#parser-objects):

By default, lxml.etree uses a standard parser with a default setup. If you want to configure the parser, you can create a new instance:

```
>>> parser = etree.XMLParser(remove_blank_text=True)  # lxml.etree only!
```

This creates a parser that removes empty text between tags while parsing, which can reduce the size of the tree and avoid dangling tail text if you know that whitespace-only content is not meaningful for your data. An example:

```
>>> root = etree.XML("<root>  <a/>   <b>  </b>     </root>", parser)

>>> etree.tostring(root)
b'<root><a/><b>  </b></root>'
```

Note that the whitespace content inside the `<b>` tag was not removed, as content at leaf elements tends to be data content (even if blank). You can easily remove it in an additional step by traversing the tree:

```
>>> for element in root.iter("*"):
...     if element.text is not None and not element.text.strip():
...         element.text = None

>>> etree.tostring(root)
b'<root><a/><b/></root>'
```

So - we can remove it as suggested in the docs, but I am not convinced we should do it.

Reply:
Be careful, I'm not sure about it but if we have elements that contain e.g. test output, like `<failure/>`, it may contain whitespace and it might be expected. A post-pocessing like the one mentioned above might harm elements like this.

Reply:
>  it may contain whitespace and it might be expected

Yes, exactly. Let's keep it as it is now.

Reply:
@seberm thanks, what can we do, let's continue with what we have, it is still valid xml :)

Thanks for chekcking